### ISSUE TITLE
No linter error reported for invalid syntax: unparenthesized assignment expression in assert.

### ISSUE THREAD

**Issue by BeefaloKing** [microsoft/pyright#3845]
## Environment data

-   Language Server version: v2022.8.30
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.6
-  "pyhton.analysis.typeCheckingMode": "strict"

## Code Snippet

```python
assert foo := 'bar'
```

## Expected behavior
Since an unparenthesized assignment expression is not allowed in this context (see https://github.com/python/cpython/issues/84090), I should see one or more linter errors suggesting as such similar to when assignment expressions are incorrectly used in other contexts such as:
```python
foo := 'bar'
```

## Actual behavior
No linter errors are reported and `foo` now shows as `(variable) foo: Literal['bar']`.

**Comment by erictraut** [comment]
Thanks for the bug report. [PEP 572](https://peps.python.org/pep-0572/) didn't call out the behavior of expressions used within `assert` statements, so I'm surprised to see that the implementers chose to preclude the use of unparenthesized assignment expressions in this context. The semantics seem clear and unambiguous in this context, so this behavior strikes me as inconsistent.

In any case, the CPython implementation is ultimate arbiter of correctness here, so I'll update pyright's parser to match the runtime behavior.

**Comment by erictraut** [comment]
This will be addressed in the next release of pyright and included in a future release of pylance.

Here's the commit: https://github.com/microsoft/pyright/commit/341af785aa189ba39e0be60f7dc65231fcd1acf5

**Comment by BeefaloKing** [comment]
> The semantics seem clear and unambiguous in this context

No kidding. Thanks for the quick turn around.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.268, which I just published. It will also be included in a future version of pylance.
